Enable configuring certificate keystore type

ntallar opened the pull request:
## Description

Enables configuring the type of certificate keystore being used.

This allows recreating a certificate keystore from a certificate and key files (using `openssl pkcs12 -export ...` command), which is needed for obtaining a keystore for the Mantis client based on the `server.crt` and `server.key` from [Daedalus](https://github.com/input-output-hk/daedalus/tree/master/tls).

rtkaczyk commented on src/main/resources/application.conf:
```diff
@@ -155,17 +155,21 @@ mantis {
-      # null value indicates no certificate is being used
+      # Type of certificate keystore being used
+      # null value indicates HTTPS is not being used
+      certificate-keystore-type = null
```
What are valid keystore types? Maybe that info could be included in the comment here?

rtkaczyk replied:
Detail: a clarification was added to `network.conf`. Could we also add it to this file `application.conf`?

ntallar commented:
> Any setup guidance anywhere?

The steps required for using this for setting this up for using Daedalus were added to the related [Youtrack ticket](https://iohk.myjetbrains.com/youtrack/issue/EC-339), I just updated (and moved some of the setup from the ticket) to the [configuring HTTPS wiki](https://github.com/input-output-hk/mantis/wiki/Configuring-HTTPS-for-JSON-RPC).
